packages/terra-clinical-result/src/flowsheet-result-cell/FlowsheetResultCell.jsx

```diff
@@ -303,11 +303,33 @@ const createFlowsheetResultCellDisplay = (resultDataSet, hideUnit, numericOverfl
     });
     const displayCount = additionalResultCount + 1;
     const hasAccessoryIcons = (firstResultAttributes.comment || firstResultAttributes.modified || firstResultAttributes.unverified);
-    const additionalResultsStackDisplay = createEndAdditionalResultsStack(displayCount, additionalResultInterpretations, hasAccessoryIcons, resultKeyID);
+    const additionalResultsStack = createEndAdditionalResultsStack(displayCount, additionalResultInterpretations, hasAccessoryIcons, resultKeyID);
+
+    if (hasAccessoryIcons) {
```

Comment:
Is this grouping needed? The code below adds additionalResultsStack and endAccessoryIcons separately. What scenario is grouping for both needed vs separate results stack and accessory icons?

Reply:
I did it like this because when there is an end accessory icon and additional results they need to be grouped in the same parent container. This is so the order of end icons shows up correctly - we need to float the whole container to the right hand side instead of the additional results stack icon and accessory icons individually (when both are present). When it's just the additional results stack or just the accessory icons they still are put into the container so we don't have to worry in the css about floating those individually. It's a bit complicated so more info about my thought process can be found here: https://github.com/cerner/terra-clinical/pull/865#discussion_r1223572374

To break it down further: 

1. If there are additional results and accessory icons we create them, put those into a container together, push it to the composite cell and return.
2. If there are additional results but no accessory icons then it creates just the additional results, puts it into the container, pushes it, and returns. We don’t need to continue to the next bit of logic since we already checked if there were accessory icons.
3. If no additional results then we just check if there are accessory end icons, if there are then we create them, put them in the container, and push. I realized I needed to create a check for hasAccessoryIcons so that we don’t create an empty div if createEndIcons returns null. Added that logic here: https://github.com/cerner/terra-clinical/pull/865/commits/8f533dc140ddcfdd4115e688807f767be79e73d2

Reply:
Thanks for the additional info. I think this info would be good to include as a code comment.

Reply:
Added a comment, let me know if it's clear enough. Was trying to stay concise and just mention the main reason for that logic block being there: [b063110](https://github.com/cerner/terra-clinical/pull/865/commits/b063110b15c2ad1f1345416f80380b51063bfe75)